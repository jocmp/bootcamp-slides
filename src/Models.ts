export interface SlideshowModel {
    id: number;
    title: string;
    slides: SlideModel[];
}

export interface SlideModel {
    title: string;
    slide_type: string;
    content?: string[];
}

export interface AppState {
    slideshow: SlideshowModel;
}