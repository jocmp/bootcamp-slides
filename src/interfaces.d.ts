declare module 'enzyme-to-json'