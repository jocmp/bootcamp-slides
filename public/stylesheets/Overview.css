.overview {
  width: 20%;
  display: block;
  float: left;
  margin-right: 4px;
  margin-left: 2%;
  border: 1px solid #4C4845;
  height: 60vh;
  margin-top: 10px;
  text-align: left;
  padding-left: 16px; }
  .overview li {
    text-align: left; }
