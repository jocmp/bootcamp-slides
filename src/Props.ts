import { SlideModel, SlideshowModel } from './Models'

export interface SlideshowOwnProps {
    history: any;
    match: { params: any };
}

export interface SlideshowProps extends SlideshowStateProps, SlideshowDispatchProps, SlideshowOwnProps { };

export interface SlideshowStateProps {
    slideshow: SlideshowModel;
    viewedIndices: number[];
}

export interface SlideshowDispatchProps {
    handleNextSlide: (id: string, index: string, length: number, history: any) => any;
    handlePreviousSlide: (id: string, index: string, length: number, history: any) => any;
    fetchSlideshow: (id: number, index: number) => void;
    viewSlide: (id: number) => any;
}

export interface OverviewProps {
    match: { params: any };
    slideshow: SlideshowModel;
    viewedIndices: number[];
    viewSlide: (id: number) => any;
}

export interface OverviewItemProps {
    slideshow_id: number;
    index: number;
    slide: SlideModel;
    viewed: boolean;
    isCurrent: boolean;
    viewSlide: (id: number) => any;
}

export interface SlideProps {
    slide: SlideModel;
}
