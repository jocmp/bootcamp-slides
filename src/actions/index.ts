import { SlideshowModel } from '../Models'
import { History } from 'history';

export const nextSlide = (id: string, index: string, totalLength: number, history: History) => {
    const currentIndex = parseInt(index);
    const nextIndex: number = currentIndex < totalLength - 1 ? currentIndex + 1 : currentIndex;
    return handleSlide(id, nextIndex, history);
}

export const previousSlide = (id: string, index: string, totalLength: number, history: History) => {
    const currentIndex = parseInt(index);
    const previousIndex: number = currentIndex > 0 ? currentIndex - 1 : currentIndex;
    return handleSlide(id, previousIndex, history);
}

const handleSlide = (id: string, index: number, history: History) => {
    history.push(`/slideshows/${id}/slides/${index}`);
    return viewSlide(index);
}

export const viewSlide = (index: number) => {
    return {
        type: 'VIEW_SLIDE',
        index
    };
}

const apiData = (index: number): SlideshowModel => {
    const shows: SlideshowModel[] = [
        {
            id: 1,
            title: "Exceptional Presentation",
            slides: [
                {
                    title: "Title here",
                    slide_type: "title"
                },
                {
                    title: "Sensational Informational",
                    slide_type: "simple",
                    content: ["Inspirational insatiable"]
                }
            ]
        },
        {
            id: 2,
            title: "Touch Fuzzy, Get Dizzy",
            slides: [
                {
                    title: "Yoshi's Island",
                    slide_type: "title",
                },
                {
                    title: "Yoshi's Island",
                    slide_type: "simple",
                    content: ['Imperically the best Super Mario game... \ntest']
                },
                {
                    title: "Why buy a SNES?",
                    slide_type: "simple",
                    content: ["Yoshi's Island. No question."]
                }
            ]
        }
    ]
    return shows[index - 1];
};

export const loadSlideshow = (slideshow: SlideshowModel, currentIndex: number) => {
    return {
        type: 'LOAD_SLIDESHOW',
        slideshow,
        currentIndex
    }
};

export const fetchSlideshow = (id: number, index: number = 0) => (dispatch: any, getState: any) => {
    dispatch(loadSlideshow(apiData(id), index));
};