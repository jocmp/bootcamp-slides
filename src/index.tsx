import * as React from 'react';
import * as ReactDOM from 'react-dom';
import App from './components/App';
import './index.css';
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

export const slideshow = (state: any, action: any) => {
  switch(action.type) {
    case 'LOAD_SLIDESHOW':
      return {
        slideshow: action.slideshow
      }
    default:
      return {};
  }
}

export const store = createStore(slideshow, applyMiddleware(thunk));
store.subscribe(() => console.log(store.getState()))

ReactDOM.render(
  <App store={store} />,
  document.getElementById('root') as HTMLElement
);
