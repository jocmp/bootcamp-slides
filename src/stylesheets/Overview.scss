@import "./colors";

.overview {
    width: 20%;
    display: block;
    float: left;
    margin-right: 4px;
    margin-left: 2%;
    border: 1px solid $black;
    height: 60vh;
    margin-top: 10px;
    text-align: left;
    padding-left: 16px;

    li {
        text-align: left;
    }
}

.overview-item-viewed {
    font-weight: bold;
}