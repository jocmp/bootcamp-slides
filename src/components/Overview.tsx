import * as React from 'react'
import { OverviewProps } from '../Props'
import { SlideModel, SlideshowModel } from '../Models'
import OverviewItem from './OverviewItem'
import '../stylesheets/Overview.scss'

const getSlides = (slideshow: SlideshowModel): SlideModel[] => {
    if (!slideshow || !slideshow.slides) {
        return [];
    }
    return slideshow.slides;
};

const Overview = (props: OverviewProps) => {
    const slides = getSlides(props.slideshow);
    return (
        <div className="overview">
            <h4>Overview</h4>
            <ul>
                {slides.map((slide, index) =>
                    <OverviewItem viewSlide={props.viewSlide} viewed={props.viewedIndices.indexOf(index) >= 0} key={index} slideshow_id={props.slideshow.id} index={index} slide={slide} />
                )}
            </ul>
        </div>
    )
}

export default Overview;