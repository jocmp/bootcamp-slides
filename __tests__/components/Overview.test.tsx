import * as React from 'react'
import Overview from '../../src/components/Overview'
import { SlideshowModel, SlideModel } from '../../src/Models'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Overview', () => {

    const slide: SlideModel = {
        title: "Exciting Slide",
        slide_type: "title",
    }; 
    const show: SlideshowModel = {
        id: 1,
        title: "Exciting Presentation",
        slides: [ slide, slide, slide ]
    };

    it('renders', () => {
        const wrapper = shallow(
            <Overview slideshow={show} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
}); 