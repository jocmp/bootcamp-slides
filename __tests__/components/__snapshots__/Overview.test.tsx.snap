// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders 1`] = `
<div
  className="overview"
>
  <h4>
    Overview
  </h4>
  <ul>
    <OverviewItem
      index={0}
      slide={
        Object {
          "slide_type": "title",
          "title": "Exciting Slide",
        }
      }
      slideshow_id={1}
      viewed={false}
    />
    <OverviewItem
      index={1}
      slide={
        Object {
          "slide_type": "title",
          "title": "Exciting Slide",
        }
      }
      slideshow_id={1}
      viewed={false}
    />
    <OverviewItem
      index={2}
      slide={
        Object {
          "slide_type": "title",
          "title": "Exciting Slide",
        }
      }
      slideshow_id={1}
      viewed={false}
    />
  </ul>
</div>
`;
